dig/README.md

```diff
@@ -0,0 +1,78 @@
+# Dependency Injection Graph
+
+This package provides a fairly opinionated way of resolving object dependencies.
+There are two sides of dig: `Inject` and `Resolve`
+
+## Inject
+
+Adding an object, or a constructor of an object to the graph.
+
+There are two ways of providing an object:
+
+1. Injecting a pointer to an existing object
+1. Injecting a "constructor function" that returns one pointer (or interface)
+
+### Injecting an object
+
+Injecting an object means it has no dependencies, and will be used as a
```

Comment:
"means it has no dependencies" could mean:
- it's a requirement for the object to have no dependencies (i.e., you can only inject objects without dependencies)
- removes its dependencies (i.e., when you inject an object, that object drops any dependencies)
- doesn't add any dependencies (i.e., just because you injected the object into this graph doesn't mean it's useful yet—you still need to tie it to other things)